src/tools/spake2p/Cmd_GenVerifier.cpp

```diff
@@ -0,0 +1,259 @@
+    case 'p':
+        // Specifications sections 5.1.1.6 and 5.1.6.1
+        if (!ParseInt(arg, gPinCode) || (gPinCode > chip::kSetupPINCodeMaximumValue) || (gPinCode == 0) || (gPinCode == 11111111) ||
+            (gPinCode == 22222222) || (gPinCode == 33333333) || (gPinCode == 44444444) || (gPinCode == 55555555) ||
+            (gPinCode == 66666666) || (gPinCode == 77777777) || (gPinCode == 88888888) || (gPinCode == 99999999) ||
+            (gPinCode == 12345678) || (gPinCode == 87654321))
+        {
+            PrintArgError("%s: Invalid value specified for pin-code parameter: %s\n", progName, arg);
+            return false;
+        }
+        break;
+
+    case 'i':
+        if (!ParseInt(arg, gIterationCount) ||
+            !(gIterationCount >= chip::kPBKDFMinimumIterations && gIterationCount <= chip::kPBKDFMaximumIterations))
+        {
+            PrintArgError("%s: Invalid value specified for the iteration-count parameter: %s\n", progName, arg);
+            return false;
+        }
+        break;
+
+    case 's':
+        gSalt = arg;
+        if (!(strlen(gSalt) >= chip::kPBKDFMinimumSaltLen && strlen(gSalt) <= chip::kPBKDFMaximumSaltLen))
+        {
+            fprintf(stderr, "%s: Invalid legth of the specified salt parameter: %s\n", progName, arg);
+            return false;
+        }
+        break;
+    case 'o':
+        gOutFileName = arg;
+        break;
+    default:
+        PrintArgError("%s: Unhandled option: %s\n", progName, name);
+        return false;
+    }
+
+    return true;
+}
+
+} // namespace
+
+bool Cmd_GenVerifier(int argc, char * argv[])
+{
+    FILE * outFile = NULL;
+
+    if (argc == 1)
+    {
+        gHelpOptions.PrintBriefUsage(stderr);
+        return true;
+    }
+
+    bool res = ParseArgs(CMD_NAME, argc, argv, gCmdOptionSets);
+    VerifyOrReturnError(res, false);
+
+    if (gIterationCount == 0)
+    {
+        fprintf(stderr, "Please specify the iteration-count parameter.\n");
+        return false;
+    }
+
+    if (gSalt == nullptr)
+    {
+        fprintf(stderr, "Please specify the salt parameter.\n");
+        return false;
+    }
+
+    if (gOutFileName == nullptr)
+    {
+        fprintf(stderr, "Please specify the output file name, or - for stdout.\n");
+        return false;
+    }
+
+    if (strcmp(gOutFileName, "-") != 0)
+    {
+        outFile = fopen(gOutFileName, "w+b");
+        if (outFile == NULL)
+        {
+            fprintf(stderr, "Unable to create file %s\n%s\n", gOutFileName, strerror(errno));
+            return false;
+        }
+    }
+    else
+    {
+        outFile = stdout;
+    }
+
+    if (fprintf(outFile, "Index,Iteration Count, Salt, PIN Code, Verifier\n") < 0 || ferror(outFile))
+    {
+        fprintf(stderr, "Error writing to output file: %s\n", strerror(errno));
+        return false;
+    }
+
+    for (uint32_t i = 0; i < gCount; i++)
+    {
+        chip::PASEVerifier verifier;
+        CHIP_ERROR err = chip::PASESession::GeneratePASEVerifier(
```

Comment:
We should generate random salts after the first.

Reply:
done. Added randomization option for salt parameters